src/node.rs

```diff
@@ -1143,9 +1171,13 @@ impl<S: ForkSource + std::fmt::Debug> InMemoryNode<S> {
         inner.block_hashes.insert(current_miniblock, block.hash);
         inner.blocks.insert(block.hash, block);
         {
-            inner.current_timestamp += 1;
+            // With the introduction of 'l2 blocks' (and virtual blocks),
+            // we are adding one l2 block at the end of each batch (to handle things like remaining events etc).
+            // That's why here, we increase the batch by 1, but miniblock (and timestamp) by 2.
+            // You can look at insert_fictive_l2_block function in VM to see how this fake block is inserted.
             inner.current_batch += 1;
-            inner.current_miniblock = current_miniblock;
+            inner.current_miniblock += 2;
```

Comment:
See other thread for how/why this will cause problems with `eth_blockNumber` and `eth_getBlockByNumber(x)` (where `x` is the number returned from `eth_blockNumber`)

Reply:
I've seen this - and with the virtual blocks that were added in this VM version, we might need to change some things. Let's discuss offline.

Reply:
Handled by adding 2 blocks, where the second one is empty.